core/src/main/java/com/amplifyframework/hub/HubCategory.java

```diff
@@ -90,4 +98,12 @@ public CategoryType getCategoryType() {
 
         return subscribe(channel, event -> true, transformingListener);
     }
+
+    private HubPlugin<?> getHubPlugin() {
+        if (!super.isConfigured() || super.getPlugins().isEmpty()) {
+            return defaultPlugin;
+        } else {
+            return super.getSelectedPlugin();
+        }
+    }
```

Comment:
nice. I only found out that hub plugins are supposed to be available pre-configuration in that meeting, so I didn't account for that.